lib/src/multires_stochastic_texture_synthesis.rs

```diff
@@ -1366,12 +1367,10 @@ impl TreeGrid {
         ];
         // Note locking all of them at different times seems to be the best way
         // Naively trying to lock all at once could easily result in deadlocks
-        let mut tmp_result: Vec<(i32, i32, i64)> = Vec::with_capacity(k * 9);
+        let mut tmp_result: BinaryHeap<(i64, i32, i32)> = BinaryHeap::with_capacity(k);
```

Comment:
Now that I think about it we could do this in linear time O(k) instead of O(k logk) by using the fact that the lists we get from the rtrees are already sorted. Not sure how much of a speed up it is though but I'll test that

Reply:
No worries, I will probably be too busy today to have time for this to be honest. :slightly_smiling_face:

Reply:
👍 take your time!